internal/components/receivers/k8sobjects.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receivers
+
+import (
+	"github.com/go-logr/logr"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+type k8sobjectsConfig struct {
+	Objects []k8sObject `yaml:"objects"`
+}
+
+type k8sObject struct {
+	Name  string `yaml:"name"`
+	Mode  string `yaml:"mode"`
+	Group string `yaml:"group,omitempty"`
+}
+
+func generatek8sobjectsRbacRules(_ logr.Logger, config k8sobjectsConfig) ([]rbacv1.PolicyRule, error) {
+	// The k8s Objects Receiver needs get permissions on the following resources always.
```

Comment:
Comment left over from k8sevents? We could replace it with a link to the documentation specifying why we require these specific RBAC rules.